### PR TITLE
API additions to allow pywin32-ctypes to work with pyinstaller

### PR DESCRIPTION
**Pull request by virtuald**
Our goal is to use pyinstaller in msys2 environment, and cffi/pywin32 don't work in msys2 so ctypes is our primary focus.

* Adds minimal set of APIs to allow pyinstaller to work correctly
* The ctypes api has been used successfully with pyinstaller, the cffi stuff I threw in for completeness

CC @dangmai

### CODE REVIEW

File: win32ctypes/core/cffi/_kernel32.py [hunk 1]
```diff
@@ -66,6 +74,18 @@ def _GetACP():
     return kernel32.GetACP()
 
 
+def _GetWindowsDirectory():
+    buffer = ffi.new('wchar_t[256]')
```
Comment (itziakos) [inline win32ctypes/core/cffi/_kernel32.py]: Although I am not sure at the moment how to get that value without using ctypes

Comment (virtuald) [inline win32ctypes/core/cffi/_kernel32.py]: That was my concern also. However, MAX_PATH has been the same value for 20 years, I suspect it isn't changing (and, I got it wrong -- it's actually 260).

**Comment by virtuald** [comment]
Oh weird. No wonder the tests failed for me originally, it appears that your setup.py has `use_2to3=True`.

File: win32ctypes/core/cffi/_kernel32.py (hunk 1 above)
Comment (itziakos) [inline win32ctypes/core/cffi/_kernel32.py]: if we can factor out a constant for `MAX_PATH` to be used for the cffi backend, that would be ideal in this case.

File: win32ctypes/tests/test_win32api.py [hunk 2]
```diff
@@ -130,6 +130,13 @@ def _id2str(self, type_id):
         else:
```
Comment (itziakos) [inline win32ctypes/tests/test_win32api.py]: opened issue #48 to track this

**Pull request merged**